### PR TITLE
Adding default coordinates to Datasets with missing image coords

### PR DESCRIPTION
**Pull request by jlstevens**
This PR aims to address #603 by generating default image coordinates when they are missing.

Before the PR, using the example given in #603:

With the PR:

So far I've only thought about the `Dataset` branch, to finish this PR I'll need to handle the `DataArray` case as well.

### CODE REVIEW

**Comment by jlstevens** [comment]
Some datashader related tests seem to be failing on Python 3. Python 2 is passing because:

> test_plot_resolution_0_image (hvplot.tests.testoperations.TestChart2D) ... SKIP: xarray or datashader not available

**Comment by jlstevens** [comment]
The datashader test failures are not being caused by this PR so that is a separate issue to fix. Restoring the original commit.

**Comment by rabernat** [comment]
Amazing, thanks for your work on this!

I would make sure to test many possible combinations here (no coordinates, some coordinates, dask, no dask, different ndims, etc.) The example I reported in #603 was just one fail case. Every different scenario seemed to fail with a slightly different error message, and unfortunately I did not take the time to document all of them.

File: hvplot/converter.py [hunk 1]
```diff
@@ -1026,6 +1026,12 @@ def __call__(self, kind, x, y):
             elif self.datatype == 'xarray':
                 import xarray as xr
                 if isinstance(data, xr.Dataset):
+                    if kind == 'image':
+                        dims_with_coords = set(data.coords.keys())
+                        missing_coords =  set(self.indexes) - dims_with_coords
+                        for missing_coord in missing_coords:
+                            new_coord = np.arange(data[missing_coord].shape[0])
+                            data = data.assign_coords(**{missing_coord:new_coord})
```
Comment (philippjfr) [inline hvplot/converter.py:1034]:
```suggestion
                        missing_coords =  set(self.indexes) - dims_with_coords
                        new_coords = {coord: np.arange(data[coord].shape[0])
                                                  for coord in missing_coords}
                        data = data.assign_coords(**new_coords)
```

File: hvplot/converter.py [hunk 2]
```diff
@@ -1026,6 +1026,12 @@ def __call__(self, kind, x, y):
             elif self.datatype == 'xarray':
                 import xarray as xr
                 if isinstance(data, xr.Dataset):
+                    if kind == 'image':
```
Comment (philippjfr) [inline hvplot/converter.py:1029]: How about quadmeshes?

Comment (philippjfr) [reply hvplot/converter.py:1029]: And contours and contourf?

Comment (jlstevens) [reply hvplot/converter.py:1029]: Yup, all need to be handled. PR is still WIP and not ready for review.

**Comment by jlstevens** [comment]
> I would make sure to test many possible combinations here (no coordinates, some coordinates, dask, no dask, different ndims, etc.) 

Absolutely! I'll make sure there are unit tests for these cases.

File: setup.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def get_setup_version(reponame):
     'bokeh >=1.0.0',
     'colorcet >=2',
     'holoviews >=1.11.0',
-    'pandas',
+    'pandas <=1.2.2',
```
Comment (philippjfr) [inline setup.py:34]:
```suggestion
    'pandas',
```

**Comment by philippjfr** [comment]
Going to merge this as a first cut, please do follow up on all the other cases but I'm releasing 0.7.2 tonight so wanted to get this in.

**Pull request merged**